pkg/registry/servicecatalog/broker/validation.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
```

Comment:
I'd like to see the kubelet code. For reference. 

I'd like to export as little as possible. I did not see any external usage of the many validators I checked, so I figured we could fold it in here and not export something people shouldn't be using. How about we leave a note in `apis/servicecatalog/validation` pointing them here? 

I agree that validations are closely related to the object definitions, which is why the different strategies have `Validate` functions. I can fold the validate code directly inline and then validate.go goes away! I think pretty much everything in `pkg/registry/servicecatalog` should be in `pkg/apis/servicecatalog`. I don't see any reason for the current split besides convention.